sources/engine/Stride.Input/Windows/InputSourceWindowsDirectInput.cs

```diff
@@ -116,6 +134,37 @@ public void OpenDevice(DeviceInstance deviceInstance)
                 RegisterDevice(controller);
             }
         }
+
+        private bool IsXInputDevice(Guid productGuid)
+        {
+            // Set security level to IMPERSONATE
+
+            DComSessionOptions DComOptions = new DComSessionOptions();
+            DComOptions.Impersonation = ImpersonationType.Impersonate;
+
+            var mySession = CimSession.Create(null, DComOptions);
```

Comment:
I believe all above and including this line has to be moved to `GetAllXInputDevices()`